### PR TITLE
Implement core Ui and Command Parsing

### PR DESCRIPTION
**Pull request by 3m0W33D**

### CODE REVIEW

**Review by kelvneo — APPROVED** [review]
LGTM

File: src/main/java/terminus/ui/Ui.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    public void printSection(String... strings) {
+        Arrays.stream(strings).forEach(System.out::println);
+    }
+
+    /**
+     * Prints the exit message.
+     */
+    public void printExitMessage() {
+        System.out.println("Goodbye!");
```
Comment (LouisLouis19) [inline src/main/java/terminus/ui/Ui.java:61]: might consider using a constant? 
Just to ensure consistency with the opening message

File: src/main/java/terminus/Terminus.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        this.parser = MainCommandParser.getInstance();
+        this.workspace = "";
+        this.ui.printBanner();
+        this.ui.printParserBanner(this.parser);
+    }
+    
+    private void runCommandsUntilExit() {
+        while (true) {
+            String input = ui.requestCommand(workspace);
+
```
Comment (LouisLouis19) [inline src/main/java/terminus/Terminus.java:43]: might consider removing whitespace?

File: src/main/java/terminus/command/TestCommand.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package terminus.command;
+
+import terminus.module.NusModule;
+import terminus.ui.Ui;
+
+public class TestCommand extends Command {
+    private String arguments;
```
Comment (LouisLouis19) [inline src/main/java/terminus/command/TestCommand.java:7]: please add an empty line to ensure consistency with the other classes

**Review by Woolicious98 — APPROVED** [review]

File: src/main/java/terminus/parser/CommandParser.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        this.workspace = workspace;
+        addCommand("exit", new ExitCommand());
+    }
+    
+    public Command parseCommand(String command) throws InvalidCommandException {
+        String[] commandLine = command.strip().split(SPACE_DELIMITER,2);
+        String cmdName = commandLine[0];
+        Command currentCommand = commandMap.get(cmdName.strip().toLowerCase(Locale.ROOT));
+        if (currentCommand == null) {
+            throw new InvalidCommandException("Command not found! Type 'help' for a list of commands.");
```
Comment (LouisLouis19) [inline src/main/java/terminus/parser/CommandParser.java:27]: shall we convert all messages to constants to ensure consistency with the Ui class?

**Pull request merged**